### PR TITLE
[IMP] hr: convert tests to hoot

### PR DESCRIPTION
**Pull request by ramh-odoo**
This PR aims to convert QUnit tests which rely on hr/test_utils to hoot.

Part of Task-3818666

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by rapr-odoo — CHANGES_REQUESTED** [review]
IMO we don't need to use `mail_test_helpers` here. We could manage it without those and use `web_test_helpers`. So, we could save ourselves from this trouble in future if we change something in `mail_test_helpers`.

File: addons/hr/static/tests/profile_form_view.test.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                <field name="name"/>
+                <field name="lang"/>
+            </form>`,
+        resId: userId,
+    });
+    patchWithCleanup(form.env.services.action, {
+        doAction: (action, options) => {
+            expect.step(action);
+        },
+    });
```
Comment (rapr-odoo) [inline addons/hr/static/tests/profile_form_view.test.js:34]: Instead you could use `mockService` before you mount view.

File: addons/hr/static/tests/web/m2x_avatar_user.test.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        return {
+            registerToImStatus() {},
+            unregisterFromImStatus() {},
+        };
+    },
+};
+
+test("avatar card preview with hr", async () => {
+    mockService("multi_tab", fakeMultiTab);
+    mockService("im_status", fakeImStatusService);
```
Comment (rapr-odoo) [inline addons/hr/static/tests/web/m2x_avatar_user.test.js:47]: Is it necessary to do this?

File: addons/hr/static/tests/web/m2x_avatar_user.test.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    const { env } = await makeMockServer();
+    const departmentId = env["hr.department"].create({
+        name: "Management",
+    });
```
Comment (rapr-odoo) [inline addons/hr/static/tests/web/m2x_avatar_user.test.js:51]: Could be done in single line.

File: addons/hr/static/tests/web/m2x_avatar_user.test.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        ]);
+        return [
+            {
+                name: "Mario",
+                job_title: "sub manager",
+                department_id: [departmentId, "Management"],
+                work_email: "Mario@odoo.pro",
+                work_phone: "+585555555",
+            },
+        ];
```
Comment (rapr-odoo) [inline addons/hr/static/tests/web/m2x_avatar_user.test.js:85]: Why need to return static data?

File: addons/hr/static/tests/web/m2x_avatar_user.test.js [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        resModel: "m2x.avatar.user",
+        arch: `<kanban>
+        <templates>
+            <t t-name="kanban-box">
+                <div>
+                    <field name="user_id" widget="many2one_avatar_user"/>
+                </div>
+            </t>
+        </templates>
+    </kanban>`,
```
Comment (rapr-odoo) [inline addons/hr/static/tests/web/m2x_avatar_user.test.js:98]: Indentation.

File: addons/hr/static/tests/web/m2x_avatar_user.test.js [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    });
+    await contains(queryFirst(".o_m2o_avatar > img")).click();
+    expect.verifySteps(["user read"]);
+    expect(".o_avatar_card").toHaveCount(1);
+    expect(queryAllTexts(queryAll(".o_card_user_infos > *"))).toEqual([
```
Comment (rapr-odoo) [inline addons/hr/static/tests/web/m2x_avatar_user.test.js:103]: I don't think `queryAll` is necessary here.

File: addons/hr/static/tests/legacy/helpers/model_definitions_setup.js [hunk 7]
```diff
@@ -1,10 +1,5 @@
...
 
-import { addFakeModel, addModelNamesToFetch } from '@bus/../tests/helpers/model_definitions_helpers';
+import {  addModelNamesToFetch } from '@bus/../tests/helpers/model_definitions_helpers';
 
 addModelNamesToFetch(['hr.employee', 'hr.employee.public', 'hr.department']);
-
-addFakeModel('m2x.avatar.employee', {
-    employee_id: { string: "Employee", type: 'many2one', relation: 'hr.employee.public' },
-    employee_ids: { string: "Employees", type: "many2many", relation: 'hr.employee.public' },
-});
```
Comment (rapr-odoo) [inline addons/hr/static/tests/legacy/helpers/model_definitions_setup.js:10]: Is this file really needed now that you have updated all the tests?

**Review by rapr-odoo — CHANGES_REQUESTED** [review]

File: addons/hr/static/tests/legacy/helpers/model_definitions_setup.js (hunk 7 above)
Comment (ramh-odoo) [reply addons/hr/static/tests/legacy/helpers/model_definitions_setup.js:10]: I think this file is still required, as the test cases for many other modules have not yet been converted to Hoot and still rely on these models.

**Comment by rapr-odoo** [comment]
We should re-target it to Master instead. As, it is not worth it do go through so much trouble of forward-ports.

**Comment by ramh-odoo** [comment]
#179716
